### PR TITLE
Optimized Slasher Should Use its Own DB File

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

Slasher has significant pressure on the beacon node's database, but it doesn't particularly require data specific to the beaconDB. Instead, we use a separate DB instance for slasher, in a separate file next to the beaconchain.db called slasher.db.

No new logic is introduced, only moving slasher-specific db files and functions into their own package and initializing a separate slasher db from the beacon db at runtime if slasher is enabled

### CODE REVIEW

File: beacon-chain/db/slasherkv/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "kv.go",
+        "metrics.go",
+        "schema.go",
+        "slasher.go",
+    ],
+    importpath = "github.com/prysmaticlabs/prysm/beacon-chain/db/slasherkv",
+    visibility = ["//visibility:public"],
```
Comment (prestonvanloon) [inline beacon-chain/db/slasherkv/BUILD.bazel:13]: Please limit to appropriate viz

File: beacon-chain/db/iface/interface.go [hunk 2]
```diff
@@ -143,6 +106,48 @@ type HeadAccessDatabase interface {
 	EnsureEmbeddedGenesis(ctx context.Context) error
 }
 
+// SlasherDatabase interface for persisting data related to detecting slashable offenses on eth2.
```
Comment (terencechain) [inline beacon-chain/db/iface/interface.go:109]: Maybe in a different file, but do this in another PR

**Review by terencechain — APPROVED** [review]
